File: projects/Ляшук/lab7/app/src/main/java/com/example/lab7/VideoActivity.java
package com.example.lab7;

import android.net.Uri;
import android.os.Bundle;
import android.widget.Button;
import android.widget.VideoView;
import android.widget.MediaController;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class VideoActivity extends AppCompatActivity {
    private VideoView videoView;
    private DatabaseHelper dbHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_video);

        dbHelper = new DatabaseHelper(this);
        dbHelper.addRecord("Открыта активность Видео (Ляшук В.И.)");

        videoView = findViewById(R.id.videoView);
        Button btnPlay = findViewById(R.id.btnPlayVideo);
        Button btnPause = findViewById(R.id.btnPauseVideo);
        Button btnBack = findViewById(R.id.btnBackVideo);

        MediaController mediaController = new MediaController(this);
        mediaController.setAnchorView(videoView);
        videoView.setMediaController(mediaController);

        // Инициализация видео. Файл должен быть в res/raw/video_sample.mp4
        int resId = getResources().getIdentifier("video_sample", "raw", getPackageName());
        if (resId != 0) {
            String videoPath = "android.resource://" + getPackageName() + "/" + resId;
            videoView.setVideoURI(Uri.parse(videoPath));
        } else {
            Toast.makeText(this, "Файл video_sample не найден в res/raw!", Toast.LENGTH_LONG).show();
        }

        btnPlay.setOnClickListener(v -> {
            videoView.start();
            Toast.makeText(this, "Ляшук: Видео запущено", Toast.LENGTH_SHORT).show();
            dbHelper.addRecord("Воспроизведение видео запущено (Ляшук В.И.)");
        });

        btnPause.setOnClickListener(v -> {
            if (videoView.isPlaying()) {
                videoView.pause();
                dbHelper.addRecord("Видео на паузе");
            }
        });

        btnBack.setOnClickListener(v -> finish());
    }
}

File: projects/Гончеренок/lab5/src/MediaApp/app/src/main/java/com/example/mediaapp/VideoActivity.kt
package com.example.mediaapp

import android.net.Uri
import android.os.Bundle
import android.widget.Button
import android.widget.VideoView
import androidx.appcompat.app.AppCompatActivity

class VideoActivity : AppCompatActivity() {

    private lateinit var videoView: VideoView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_video)

        title = "Video — Гончерёнок К.А. АС-66"

        videoView = findViewById(R.id.videoView)

        val btnStart = findViewById<Button>(R.id.btnStartVideo)
        val btnPause = findViewById<Button>(R.id.btnPauseVideo)
        val btnStop = findViewById<Button>(R.id.btnStopVideo)
        val btnBack = findViewById<Button>(R.id.btnBackVideo)

        val videoUri = Uri.parse("android.resource://$packageName/${R.raw.my_video}")
        videoView.setVideoURI(videoUri)

        btnStart.setOnClickListener {
            videoView.start()
        }

        btnPause.setOnClickListener {
            if (videoView.isPlaying) {
                videoView.pause()
            }
        }

        btnStop.setOnClickListener {
            videoView.stopPlayback()
            videoView.setVideoURI(videoUri)
        }

        btnBack.setOnClickListener {
            finish()
        }
    }

    override fun onPause() {
        super.onPause()
        if (videoView.isPlaying) {
            videoView.pause()
        }
    }
}